src/display/editor/tools.js

```diff
@@ -468,6 +501,27 @@ class AnnotationEditorUIManager {
     layer?.onTextLayerRendered();
   }
 
+  #addKeyboardManager() {
+    // The keyboard events are caught at the container level in order to be able
+    // to execute some callbacks even if the current page doesn't have the
+    // focus.
+    this.#container?.addEventListener("keydown", this.#boundKeydown);
```

Comment:
Here, and similarly below, I'm not sure that the `container` can actually be undefined (since it's a required parameter)?

Reply:
Ok, I was not sure, hence I preferred to be defensive.

Reply:
Given that it comes from the `BaseViewer` that should be safe, note how there's a bunch of validation which means that it simply cannot be undefined:
https://github.com/mozilla/pdf.js/blob/7a4b72ed11fdde82f3a77417ea369412f414f231/web/base_viewer.js#L252-L274